implementation of deleting from settings using modal

gruiz17 opened the pull request:
remove delete project modal html

several changes
* don't show alert for just any project in the query string
* UI fixes on modal
* rename functions for clarity

go bin

jwforres commented on assets/app/scripts/controllers/projects.js:
```diff
@@ -73,6 +73,17 @@ angular.module('openshiftConsole')
         $scope.showGetStarted = hashSizeFilter($scope.projects) === 0;
+
+        // if projectToDelete is in URL and if project exists to be deleted
+        if ($location.search()['projectToDelete']) {
```
instead of passing this as a URL parameter, I'd prefer this was just done in memory, could have a flash messages service or something

jwforres commented on assets/app/views/settings.html:
```diff
@@ -1,9 +1,26 @@
       </div>
+
+      <!-- delete project modal -->
+      <script type="text/ng-template" id="delete-project.html">
```
After seeing the modal we are thinking it would be nice if the project list delete also used the modal to confirm instead of the inline buttons.  So would probably want to make this its own template so it can be re-used in both places.

gruiz17 commented:
@sg00dwin @jwforres i'm actually going to make a few more changes

* putting the delete modal in a separate file, rather than right in the HTML
* using the delete modal for the project list view

jwforres commented:
@gruiz17 our QE team opened a bug for the error message that appears when a viewer tries to delete a project and doesn't have permissions, currently it dumps a raw API blob, should probably be pulling the error message out when we have one, see https://bugzilla.redhat.com/show_bug.cgi?id=1258720

gruiz17 commented:
@jwforres I have fixed this! I show the error message now instead of the entire error api blob.

@sg00dwin I've also refactored the projects page to use the modal instead.

gruiz17 commented:
@jwforres in my latest commit, i've also switched to using a service instead of url params when redirecting from settings to home :smile:

jwforres commented on assets/app/scripts/services/alertMessage.js:
```diff
@@ -0,0 +1,14 @@
+  .service("AlertMessageService", function(){
+    var alerts = [];
+    return {
+      addAlert: function(alert) {
```
need a way to clear the alerts or once you have added an alert to this its going to show up every time you navigate to the project list

jwforres commented on assets/app/scripts/controllers/settings.js:
```diff
@@ -18,8 +18,46 @@ angular.module('openshiftConsole')
+
+      modalInstance.result.then(function(result) {
+      /* upon clicking delete button, redirect to the home page,
+         with project name in query string to show alert that
```
Update this comment since project isn't in the query anymore

jwforres commented on assets/app/scripts/controllers/projects.js:
```diff
@@ -26,46 +28,40 @@ angular.module('openshiftConsole')
-        $timeout(function() {
-          delete $scope.alerts[projectName];
-        }, 10000);
+          DataService.list("projects", $scope, function(projects) {
```
pull these 4 lines out into a function called loadProjects and call it from here and from within the AuthService.withUser block so that we arent duplicating this

jwforres commented on assets/app/views/settings.html:
```diff
@@ -2,8 +2,17 @@
+        <div class="actions action-button pull-right">
+          <i class="fa fa-trash-o fa-2x" 
+            ng-click="openDeleteModal()"
+            ng-if="!deletingMap[project.metadata.name] && !confirmingMap[project.metadata.name]" 
```
i think this was copied by mistake, deletingMap and confirmingMap dont exist on the settings page

jwforres commented:
attach screenshots to the PR for what it looks like now

gruiz17 commented:
@jwforres i've made all the fixes based on your feedback.

here is a gif!

![project-settings-pr](https://cloud.githubusercontent.com/assets/2757082/9639501/327e9296-5161-11e5-8ea8-baf0bb3a6162.gif)

gruiz17 commented:
modal

![image](https://cloud.githubusercontent.com/assets/2757082/9639558/897b3130-5161-11e5-80ac-ca7ceea9b52b.png)

gif showing flow from projects page

![project-settings-pr-2](https://cloud.githubusercontent.com/assets/2757082/9639566/930db646-5161-11e5-9494-b7457f85d1e8.gif)

jwforres commented:
[merge]
